ControlFlowArraysAndStrings/ControlFlowArraysAndStrings/Program.cs

```diff
@@ -0,0 +1,343 @@
+        {
+            string[] newArray = new string[array.Length + 1];
+
+            newArray[index] = value;
+
+            // Copy values from old array into the new array.
+            // Skip over the newly added value.
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                if (i < index)
+                {
+                    newArray[i] = array[i];
+                }
+
+                if (i > index)
+                {
+                    newArray[i] = array[i - 1];
+                }
+            }
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Removes the value at the beginning of a specified array.
+        /// </summary>
+        /// <param name="array">The array to be changed.</param>
+        /// <returns>A new array with a value removed from the start.</returns>
+        public static string[] RemoveElementFromStart(string[] array)
+        {
+            string[] newArray = new string[array.Length - 1];
+
+            for (int i = 1; i < array.Length; i++)
+            {
+                newArray[i - 1] = array[i];
+            }
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Removes the value at the end of the specified array.
+        /// </summary>
+        /// <param name="array">The array to be changed.</param>
+        /// <returns>A new array with a value removed from the end.</returns>
+        public static string[] RemoveElementFromEnd(string[] array)
+        {
+            string[] newArray = new string[array.Length - 1];
+
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                newArray[i] = array[i];
+            }
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Removes the value at the specified index from the specified array.
+        /// </summary>
+        /// <param name="array">The array to be changed.</param>
+        /// <param name="index">The index to be removed.</param>
+        /// <returns>A new array with a value removed.</returns>
+        public static string[] RemoveElementFromAnywhere(string[] array, int index)
+        {
+            string[] newArray = new string[array.Length - 1];
+
+            for (int i = 0; i < newArray.Length; i++)
+            {
+                if (i < index)
+                {
+                    newArray[i] = array[i];
+                }
+
+                if (i >= index)
+                {
+                    newArray[i] = array[i + 1];
+                }
+            }
+
+            return newArray;
+        }
+
+        /// <summary>
+        /// Attempts to log in a user and prints a message to the console if login 
+        /// was successful or unsuccessful. Allows 3 unsuccessful attempts before lock out.
+        /// </summary>
+        public static void Login()
+        {
+            int loginAttempts = 3;
+            string[] credentialType = { "user name", "password"};
```

Comment:
This is a good attempt at generalizing (both "username" and "password" are "credentials" and so you can write logic to handle arbitrary credentials and you don't have to write two sets of similar logic) but it falls short.  You end up with some confusing code and then you have that awkward if statement in `ValidateCredential` (this sort thing is called a "leaky abstraction").

Later, when we learn OOP, or more functional concepts, there are ways you could maybe achieve the goal you set out for.